nilearn/plotting/img_plotting.py

```diff
@@ -738,3 +740,136 @@ def display_factory(display_mode):
+            the nodes in one go
+        edges_threshold: str or number
+            If it is a number only the edges with a value greater than
+            edges_threshold will be shown.
+            If it is a string it must finish with a percent sign, e.g. "25.3%"
+            and only the edges with a abs(value) above the given percentile
+            will be shown.
+        output_file : string, or None, optional
+            The name of an image file to export the plot to. Valid extensions
+            are .png, .pdf, .svg. If output_file is not None, the plot
+            is saved to a file, and the display is closed.
+        display_mode : {'ortho', 'x', 'y', 'z'}
+            Choose the direction of the cuts: 'x' - saggital, 'y' - coronal,
+            'z' - axial, 'ortho' - three cuts are performed in orthogonal
+            directions.
+        figure : integer or matplotlib figure, optional
+            Matplotlib figure used or its number. If None is given, a
+            new figure is created.
+        axes : matplotlib axes or 4 tuple of float: (xmin, ymin, width, height), optional
+            The axes, or the coordinates, in matplotlib figure space,
+            of the axes used to display the plot. If None, the complete
+            figure is used.
+        title : string, optional
+            The title dispayed on the figure.
+        annotate: boolean, optional
+            If annotate is True, positions and left/right annotation
+            are added to the plot.
+        black_bg: boolean, optional
+            If True, the background of the image is set to be black. If
+            you wish to save figures with a black background, you
+            will need to pass "facecolor='k', edgecolor='k'" to pylab's
+            savefig.
+        alpha: float between 0 and 1
+            Alpha transparency for the brain schematics
+
+    """
+    adjacency_matrix_shape = adjacency_matrix.shape
+    if (len(adjacency_matrix_shape) != 2 or
+            adjacency_matrix_shape[0] != adjacency_matrix_shape[1]):
+        raise ValueError("'adjacency_matrix' is supposed to have shape (n, n)."
+                         ' Its shape was {0}'.format(adjacency_matrix_shape))
+
+    nodes_coords_shape = nodes_coords.shape
+    if len(nodes_coords_shape) != 2 or nodes_coords_shape[1] != 3:
+        raise ValueError("'nodes_coords' should be a array with shape (n, 3). "
+                         'Its shape was {0}'.format(nodes_coords_shape))
+
+    if nodes_coords_shape[0] != adjacency_matrix_shape[0]:
+        raise ValueError(
+            "Shape mismatch between 'adjacency_matrix' "
+            "and 'nodes_coords'"
+            "'adjacency_matrix' shape is {0}, 'nodes_coords' shape is {1}"
+            .format(adjacency_matrix_shape, nodes_coords_shape))
+
+    if sparse.issparse(adjacency_matrix):
+        adjacency_matrix = adjacency_matrix.toarray()
+
+    if not is_spd(adjacency_matrix, accept_semi_definite=True):
```

Comment:
The function should not be limited to accepting SPD matrices (or correlation matrices), we want to be able to plot any graph defined by its adjacency.